clang/lib/CConv/RewriteUtils.cpp

```diff
@@ -675,6 +675,7 @@ bool TypeRewritingVisitor::VisitFunctionDecl(FunctionDecl *FD) {
     s = s + ")";
   } else {
     s = s + "void)";
+    DidAny = true;
```

Comment:
One drawback with this approach is that for functions that do have `void` for the args, you will change their formatting. For example, if we had the code:
```
int *
foo(void) {
  return (int *)5;
}
```
I have a feeling (correct me if I'm wrong) that this will get rewritten to
```
int *foo(void) {
  return (int *)5;
}
```
That is, we will drop the formatting around the prototype, even though there was no need to, since the type was fine before. We'd like to keep as much formatting as possible.

I think it would be better to confirm that a no-argument function is indeed defined in the original code without `void` before setting `DidAny` to be true. I believe that in `clang` there is a function that says whether a type is a proper prototype or not. If it is, that means it had `void`, a return type, etc. Have a look at this (and elsewhere in our tool's code)?  https://clang.llvm.org/doxygen/classclang_1_1Type.html#ae8750d7d0edc2b34fb395f79dad0fe3f

Reply:
Yes, this is a good point. In fact, in `return_not_least.c` (?), the functions already had void inside it and the rewriter added an extra space that I had to manually add. I actually tried to counteract this by getting the type of the function as a string and then processing it to see if it had "void" as its argument but abandoned it because it didn't seem robust enough. Had no clue that a handy function like `isFunctionPrototype` existed! 
I think my latest commit is what we're after. I tried it out on the code snippet you posted above and it doesn't unnecessarily convert the function :)

Reply:
Great! I’ll take a look in a little bit.